plugin-viewport/src/main/kotlin/com/mapbox/maps/plugin/viewport/state/FollowPuckViewportStateImpl.kt

```diff
@@ -126,6 +138,9 @@ internal class FollowPuckViewportStateImpl(
     checkLocationComponentEnablement()
     addIndicatorListenerIfNeeded()
     dataSourceUpdateObservers.add(viewportStateDataObserver)
+    if (shouldNotifyLatestViewportData()) {
+      notifyViewportStateDataObserver(viewportStateDataObserver, evaluateViewportData())
```

Comment:
The `observeDataSource` is currently only used for ViewportTransition, so it's important when `observeDataSource` is called, it should immediately be notified of current location and start the camera transition to the puck's location. During this transition, the ViewportState will continue sending location updates if available, so that we update the animation's target value on the fly, so it always end at the latest location at the end of the ViewportTransition.

So in the current design, it should be no issue with this behaviour.